include/exec/task.hpp

```diff
@@ -320,16 +320,16 @@ namespace exec {
 
      private:
       struct __final_awaitable {
-        static constexpr auto await_ready() noexcept -> bool {
+        constexpr auto await_ready() noexcept -> bool {
           return false;
         }
 
-        static auto await_suspend(__coro::coroutine_handle<__promise> __h) noexcept
+        auto await_suspend(__coro::coroutine_handle<__promise> __h) noexcept
           -> __coro::coroutine_handle<> {
           return __h.promise().continuation().handle();
         }
 
-        static void await_resume() noexcept {
+        void await_resume() noexcept {
         }
```

Comment:
the code is correct. these functions do not need to be passed a `this` pointer, hence the `static`. is there a way to suppress this warning locally?

Reply:
You are of course completely right. I'm actually on the fence of just disabling the "static readability" check. Thanks for merging the remainder!